src/sortStudents.ts

```diff
@@ -1,16 +1,72 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copy = students.map((item) => {
+    const temp = { ...item };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    return temp;
+  });
+
+  const result = copy.sort((a: Student, b: Student): number => {
```

Comment:
copy is already mutated 
```suggestion
 copy.sort((a: Student, b: Student): number => {
```